distribution/lib/Standard/Database/0.0.0-dev/src/DB_Table.enso

```diff
@@ -2931,9 +2934,11 @@ type DB_Table
                  table.write (enso_project.data / "example_csv_output.csv")
     @path (Widget.Text_Input display=Display.Always)
     @format Widget_Helpers.write_table_selector
-    write self path:Writable_File format:File_Format=Auto_Detect on_existing_file:Existing_File_Behavior=..Backup match_columns:Match_Columns=..By_Name on_problems:Problem_Behavior=..Report_Warning -> Nothing ! Column_Count_Mismatch | Illegal_Argument | File_Error =
-        # TODO This should ideally be done in a streaming manner, or at least respect the row limits.
-        self.read.write path format on_existing_file match_columns on_problems
+    write self path:Writable_File format=Auto_Detect on_existing_file:Existing_File_Behavior=..Backup match_columns:Match_Columns=..By_Name on_problems:Problem_Behavior=..Report_Warning -> Nothing ! Column_Count_Mismatch | Illegal_Argument | File_Error =
+        row_limit = 100001
+        materialized = self.read ..First row_limit
+        if materialized.row_count < row_limit then materialized.write path format on_existing_file match_columns on_problems else
+            Error.throw (Illegal_Argument.Error "DB_Table contains more than 100,000 rows. Use `read` to materialize first.")
```

Comment:
nitpick:

isn't it easier to use `100000` constant together with `<=` instead of the +1 here?

Reply:
no because can't tell if it had exactly 100000 rows :)

Reply:
Ooooh, right! I missed that detail. Makes sense now.

I'm tempted to still suggest:
```suggestion
        row_limit = 100000
        # Read 1 row more than limit to see if there are more rows in the table.
        materialized = self.read ..First row_limit+1
        if materialized.row_count <= row_limit then materialized.write path format on_existing_file match_columns on_problems else
            Error.throw (Illegal_Argument.Error "DB_Table contains more than 100,000 rows. Use `read` to materialize first.")
```

But not going to be stubborn about it :)